File: android/feature-record/src/main/kotlin/com/bolsillo/feature/record/ui/CategoryChip.kt
package com.bolsillo.feature.record.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.unit.dp
import com.bolsillo.designsystem.component.CategoryIconTile
import com.bolsillo.designsystem.theme.BolsilloTheme
import com.bolsillo.feature.record.R

@Composable
fun CategoryChip(
    categoryId: String?,
    labelKey: String,
    modifier: Modifier = Modifier,
    showLowConfidenceBorder: Boolean = false,
    onClick: () -> Unit = {},
) {
    val dark = BolsilloTheme.colors.isDark
    val palette = CategoryColorResolver.color(categoryId, dark)
    val border: Color? = if (showLowConfidenceBorder) BolsilloTheme.colors.warningBorder else null
    Row(
        modifier =
            modifier
                .clip(BolsilloTheme.shapes.chip)
                .background(BolsilloTheme.colors.surface)
                .let { m -> if (border != null) m.border(1.5.dp, border, BolsilloTheme.shapes.chip) else m }
                .padding(horizontal = 10.dp, vertical = 8.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        CategoryIconTile(foreground = palette.foreground, container = palette.container, glyph = " ")
        Column {
            Text(
                text = stringResource(R.string.record_category),
                color = BolsilloTheme.colors.textMuted,
                style = BolsilloTheme.typography.caption,
            )
            Text(
                text = labelKey,
                color = BolsilloTheme.colors.textPrimary,
                style = BolsilloTheme.typography.bodyValue,
            )
        }
    }
}

@Composable
fun AccountChip(
    accountName: String,
    modifier: Modifier = Modifier,
    onClick: () -> Unit = {},
) {
    Row(
        modifier =
            modifier
                .clip(BolsilloTheme.shapes.chip)
                .background(BolsilloTheme.colors.surface)
                .padding(horizontal = 10.dp, vertical = 8.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        CategoryIconTile(
            foreground = BolsilloTheme.colors.primary,
            container = BolsilloTheme.colors.primaryContainer,
            glyph = " ",
        )
        Column {
            Text(
                text = stringResource(R.string.record_account),
                color = BolsilloTheme.colors.textMuted,
                style = BolsilloTheme.typography.caption,
            )
            Text(
                text = accountName,
                color = BolsilloTheme.colors.textPrimary,
                style = BolsilloTheme.typography.bodyValue,
            )
        }
    }
}

File: ios/Packages/Bolsillo/Sources/FeatureRecord/Views/CategoryChip.swift
import SwiftUI
import BolsilloDesignSystem

public struct CategoryChip: View {
    let icon: String
    let label: String
    let fg: Color
    let container: Color
    let isLowConfidence: Bool
    @Environment(\.bolsilloTheme) var theme

    public init(icon: String, label: String, fg: Color, container: Color, isLowConfidence: Bool = false) {
        self.icon = icon
        self.label = label
        self.fg = fg
        self.container = container
        self.isLowConfidence = isLowConfidence
    }

    public var body: some View {
        HStack(spacing: theme.spacing.xs) {
            CategoryIconTile(icon: icon, fg: fg, container: container, size: .sm)
            VStack(alignment: .leading, spacing: 0) {
                Text(String(localized: "record.category", bundle: .module))
                    .font(theme.typography.caption)
                    .foregroundStyle(theme.colors.textMuted)
                Text(label)
                    .font(theme.typography.bodyValue)
                    .foregroundStyle(theme.colors.textPrimary)
            }
        }
        .padding(.horizontal, theme.spacing.base)
        .padding(.vertical, theme.spacing.sm)
        .background(theme.colors.surface)
        .overlay(
            RoundedRectangle(cornerRadius: theme.radius.chip)
                .stroke(
                    isLowConfidence ? theme.colors.warningBorder : theme.colors.outline,
                    lineWidth: 1.5
                )
        )
        .cornerRadius(theme.radius.chip)
    }
}

public struct AccountChip: View {
    let icon: String
    let label: String
    let balance: String?
    @Environment(\.bolsilloTheme) var theme

    public init(icon: String, label: String, balance: String? = nil) {
        self.icon = icon
        self.label = label
        self.balance = balance
    }

    public var body: some View {
        HStack(spacing: theme.spacing.xs) {
            Image(systemName: icon)
                .font(theme.typography.body)
                .foregroundStyle(theme.colors.primary)
                .frame(width: 28, height: 28)
                .background(theme.colors.primaryContainer)
                .cornerRadius(theme.radius.iconTileSm)
            VStack(alignment: .leading, spacing: 0) {
                Text(String(localized: "record.account", bundle: .module))
                    .font(theme.typography.caption)
                    .foregroundStyle(theme.colors.textMuted)
                Text(label)
                    .font(theme.typography.bodyValue)
                    .foregroundStyle(theme.colors.textPrimary)
            }
        }
        .padding(.horizontal, theme.spacing.base)
        .padding(.vertical, theme.spacing.sm)
        .background(theme.colors.surface)
        .overlay(
            RoundedRectangle(cornerRadius: theme.radius.chip)
                .stroke(theme.colors.outline, lineWidth: 1)
        )
        .cornerRadius(theme.radius.chip)
    }
}
